src/BoundaryConditions/update_boundary_conditions.jl

```diff
@@ -0,0 +1,16 @@
+using Oceananigans: boundary_conditions
+
+update_boundary_condition!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
```

Comment:
```suggestion
@inline update_boundary_condition!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
```

May need `@inline` all the way down here.